### PR TITLE
Support writing structured data directly, without JSON encoding

### PR DESCRIPTION
**Pull request by reachfh**
Adds a new option to output the structured log message as a map instead of encoding it as JSON.

This works better when using a logger formatter that encodes all log messages as JSON, avoiding double encoding.
See my library [logger_formatter_json](https://hex.pm/packages/logger_formatter_json).

With `logger_formatter_json`, the output is something like this:

```json
{"msg":{"status":"200","path":"/","method":"GET","timing":"1ms","client_ip":"10.10.102.99","duration_ms":1.286,"x_forwarded_port":"443","x_forwarded_proto":"https","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36","x_forwarded_for":"104.187.140.60"},"level":"info","file":"lib/uinta/plug.ex","line":"157","pid":"<0.3289.0>","request_id":"F6c-8tPDkCM2Ag4AAA4y","otel_trace_id":"659715447dcfbd23bc524d9bfc7a3992","otel_span_id":"e2e08b0504fbcc76"}
```

### CODE REVIEW

**Comment by epinault** [comment]
thank you for your contribution! I ll merge this and try to get a release later today. I will need to create a new minor and document the breaking change in the changelog

File: lib/uinta/plug.ex [hunk 1]
```diff
@@ -110,7 +110,20 @@ if Code.ensure_loaded?(Plug) do
...
+            value
+
+          {:ok, :map = value} ->
+            value
+
+          {:ok, :string = value} ->
+            value
+
+          :error ->
+            if Keyword.get(opts, :json, false), do: :json, else: :string
```
Comment (epinault) [inline lib/uinta/plug.ex:125]: That case statement seems a little complex. Why not make the list of options a MapSet that is compiled (@valid_formats). and check if this is a member? that would simplify the code

Comment (reachfh) [reply lib/uinta/plug.ex:119]: @epinault I changed the check to be more compact.
This should not be a breaking change, per se, more of a deprecation, as the old `json` option still works.

**Review by epinault — APPROVED** [review]

**Comment by epinault** [comment]
Fixed in https://github.com/podium/uinta/pull/53. And just published a new version

**Pull request merged**